src/main/java/com/gojek/beast/factory/BeastFactory.java

```diff
@@ -89,23 +97,48 @@ public BeastFactory(AppConfig appConfig, BackOffConfig backOffConfig, ProtoMappi
 
     public Sink createBigQuerySink() {
         BigQuery bq = getBigQueryInstance();
-        Sink bqSink = new BqSink(bq, TableId.of(appConfig.getDataset(), appConfig.getTable()));
+        BQResponseParser responseParser = new BQResponseParser();
+        BQErrorHandler bqErrorHandler = createOOBErrorHandler();
+        Sink bqSink = new BqSink(bq, TableId.of(appConfig.getDataset(), appConfig.getTable()), responseParser, bqErrorHandler);
         return new RetrySink(bqSink, new ExponentialBackOffProvider(backOffConfig.getExponentialBackoffInitialTimeInMs(), backOffConfig.getExponentialBackoffMaximumTimeInMs(), backOffConfig.getExponentialBackoffRate(), new BackOff()), appConfig.getMaxPushAttempts());
     }
 
+    public BQErrorHandler createOOBErrorHandler() {
+        final Storage gcsStore = getGCStorageInstance();
+        ErrorWriter errorWriter;
+        if (appConfig.isGCSErrorSinkEnabled()) {
+            final String bucketName = appConfig.getGcsBucket();
+            final String basePathPrefix = bucketName + "/" + appConfig.getGcsPathPrefix();
+            errorWriter = new GCSErrorWriter(gcsStore, bucketName, basePathPrefix);
+        } else {
+            errorWriter = new DefaultLogWriter();
+        }
+        return new OOBErrorHandler(errorWriter);
+    }
+
     private BigQuery getBigQueryInstance() {
+        return BigQueryOptions.newBuilder()
+                .setCredentials(getGoogleCredentials())
+                .setProjectId(appConfig.getGCPProject())
+                .build().getService();
+    }
+
+    private Storage getGCStorageInstance() {
+        return StorageOptions.newBuilder()
+                .setCredentials(getGoogleCredentials())
+                .setProjectId(appConfig.getGCPProject())
+                .build().getService();
+    }
+
+    private GoogleCredentials getGoogleCredentials() {
         GoogleCredentials credentials = null;
         File credentialsPath = new File(appConfig.getGoogleCredentials());
         try (FileInputStream serviceAccountStream = new FileInputStream(credentialsPath)) {
             credentials = ServiceAccountCredentials.fromStream(serviceAccountStream);
         } catch (IOException e) {
             e.printStackTrace();
```

Comment:
We could throw exception, or ensure the beast instance stops here if creds is not found

Reply:
It would stop as the caller `getBigQueryInstance()` would throw since the credentials is erroneous. This is an existing implementation.